src/Database/Connection.php

```diff
@@ -284,6 +288,46 @@ public function newQuery(): Query
         return new Query($this);
     }
 
+    /**
+     * Create a new SelectQuery instance for this connection.
+     *
+     * @return \Cake\Database\Query\SelectQuery
+     */
+    public function newSelectQuery(): SelectQuery
```

Comment:
Should we omit the `Query` suffix and just have `newSelect()` or do you prefer the type in the name?

Reply:
As I mentioned in another comment I want to make a separate `QueryBuilder` class, can just have `QueryBuilder::newSelect()` there.

Reply:
Could do `QueryFactory::select()` too :shrug: